### ISSUE TITLE
Companion forms and spellcasting

### ISSUE THREAD

**Issue by Marcloure** [mlenser/kryx-rpg-issues#277]
As discussed on Discord:

> 1. What is the current situation about casting spells? Wild Shape says that you keep you mana, know the spells of the creature, so on and on, but doesn't say that you can't cast spells if the form can't. So I suppose you can?
> 2. What is the DC for elemental spells? They use charisma to cast spells, so it's based on the caster's Charisma and the caster proficiency for the elemental  spells, and the caster's spellcasting ability for the caster's spells (like multiclassing in RAW)?

How I interpret this:

>1. I would say you can cast spells. Wild Shape says you keep your features and doesn't say you can't cast spells. But it is weird anyway, because it also says "you keep you mana and mana limit", so it feels like this shouldn't be the case if the form can't cast spells.
> 2. You use Charisma and your proficiency for the elemental spells, and your spellcasting ability for your spells. But that is totally me extrapolating from RAW multiclassing

Kryx commented: 
> I'm not 100% certain what is written is really how it should be. This somewhat depends on the dreaded spell components issue
>
> A bear has arms and a mouth, so it can likely fulfill the requirements for many spells that don't require specific words. But if we're strict on verbal and physical gestures then they couldn't

Also, about companion's balance, itamarcu said:
> I'd also want to consider - if we allow some companion-shapes to cast spells but not all of them, are we functionally making these bad options for most players?

That is one of my concerns as well. Why would someone turn into an animal if they could become an elemental which keep spells?

**Comment by NyusSong** [comment]
I think the first thing to focus on is the Somatic component. As presented in the rules, we have this: 

"Most spells require forceful gesticulation or an intricate set of gestures. If a spell requires a somatic component, the caster must have free use of at least one hand to perform these gestures."

Now we should just have to rule on whether or not a bear can do "forceful gesticulation or an intricate set of gestures". If we rule they do, they should be able to cast spells while Wild Shape. I think the ruling can be as simple as "They are Druids, therefore they can because they are Nature Spellcasters" or something like that. I would say they should be able to, just because they are Druids, or just because Companion forms are able to cast spells if the spellcaster is a Druid. Maybe not all spellcasters know how to cast a spell when wild shaped.

On the Elemental spells DC issue, I'd say just say that Charisma should be used since it's not part of the Druid but part of a theme. Since any spellcaster will have these rules, it makes sense that the creature uses Charisma.

**Comment by Paulorpribeiro** [comment]
I think what itamarcu said is key here, I will quote again

> I'd also want to consider - if we allow some companion-shapes to cast spells but not all of them, are we functionally making these bad options for most players?

If we don't want to make some forms strictly better than others (some form can cast while others cant) then we need to make this decision based entirely on balance an let the fluf to the DM/player.

Casting spells in wild shape should be made into a class feature for the naturalist, or a feat for anyone who wants it or an augment to the wild shape spell. 
The discussion about spell components here will poison the debate an will lead to things such as, can bats cast target spells, since they can't "see" further than 60 ft or can snakes cast somatic spells since they don't have hands and so on. Every imaginable form can cast uf the caster has a feature that says so becausehe retains his features, if he doesnt, the spell should say "if you dont have a feature that specifically allows vyou to cast in that form, you cant". Exception could be made for Inteligent humanoids I guess, but once the rule is broken, it starts to get complicated...Elementals could be able to cast spells of their element as standard and follow the feature rule to cast other spells.

On wild shape caster using charisma, it makes cha casting classes better at wild shaping, which I don't think is the intention. I think if the form can cast spells, it should use the caster's casting score

**Comment by mlenser** [comment]
I'll save my deep dive into this for tomorrow, but your spellcasting ability would not change as part of any transformation. You would just gain access to new spells.

**Comment by mlenser** [comment]
I thought I had ironed all the kinks out of Wild Shape and similar transformation spells, but looking back I think it's quite obvious that I haven't. These spells offer far too much. In order of Strength/problem:
1. **_Hit points._** This provides a huge buffer of hit points
2. **_Martial prowess._** The martial damage for a character link a Druid goes up quite substantially.
3. **_Utility._** You can acquire faster movement, flying speed, burrow speed, etc. You can also blend in with other beasts for scouting. All kind of utility here.
4. **_New Abilities._** You can gain new maneuvers/spells
5. Probably more that I haven't thought of.


## Lets compare Wild Shape to some other 2 mana spells

### Wild Shape
- Fly 60 feet for 1 hour/mana
- +52 hit points for 1 hour/mana
- AC either the same or reduced by 1 or 2 if you have high AC (it's 14)
- 2 new maneuvers
- Pack Tactics
- 2 attacks that deal ~12 damage each
- Magic attacks
- Advantage on Perception that relies on sight

### Levitate (Fly)
- Fly 60 feet for 5 minutes/mana

### False Life
- ~45.5 temporary hit points for 1 hour/mana

## Conclusions
- Hit points have to go
- Companions maybe need to be 1 lower resource when you shape into them
- Wild Shape has to lose/delay other options. Magic attacks for example, but that's minimal

**Comment by itamarcu** [comment]
Wild Shape should not be a self buff spell. It should be a "replacement"/"transformation" spell, where you **lose many of your primary capabilities** and gain new different ones. Half of this spell's usage is utility only: you want to become a giant spider to crawl where no one else can go, you want to become a fly on the wall that listens on conversations, you want to become an eel too slippery for the dragon to hold in its claws while flying with you.

The utility value of Wild Shape should not decrease. It's fine if every form only had one or two strong utility values (e.g. climbing speed and a web, or fast running speed and strong sense of smell).

The combat usefulness of Wild Shape - I'm alright with it decreasing. I'd prefer to just make its unique weaknesses weaker over make simple boring changes like reducing HP and DPR, but the solution will probably be a combination of multiple changes.

(I'm talking about wild shape and druids here, but really this applies to all form spells. The elemental ones, though, are much weaker from a utility perspective, so I think it would be alright to make them a bit stronger or versatile in a single form (within guidelines) than wild shape is, during combat)

I made lists of ways to fix this. I think we should mix and match several of the following (obviously not all options, and obviously none of these are final wordings. I'm just offering puzzle pieces):

### Boring ways to make the spell weaker
- Reduce all companion statistics to make this work (bad solution because Call Beast and Eidolon and other things all use the same statistics)
- When you enter your wild shape, you only gain half of the hit points of the creature
- When you deal damage as your wild shape, you only deal half as much damage
- Make wild shape require Concentration
- Make wild shape require something similar to concentration but not actually concentration (so that you can still concentrate on spells you cast as the new form)
- Wild shape gives you one less resource than the usual
- Wild shape has much worse AC and saves
- Wild shape can attack only once each turn

### Less boring ways to make the spell weaker
- Require the druid to augment the spell with additional mana to gain the benefits of one or multiple or all of the following (this augment is an *alternative* to adding "resources" and getting better stat blocks): 
- - The ability to use spells
- - The ability to use maneuvers
- - Special senses
- - Special features
- - Special attacks or multiple attacks
- - Attacks that deal magical damage
- - Not having to keep concentration (and add conc requirement)
- - Longer duration (and reduce normal duration to 1 or 5 minutes. this is good IMO, because there's a difference between long duration noncombative forms and short duration combative forms)
- - The ability to convert mana into hp
- - The ability to speak in animal form
- - The ability to retain your feats and features
- - The ability to retain your mental ability scores (including the ability to make intelligent choices)
- Change the wild shape forms so that they are all really specific in what they can do. A bear can maul and use maneuvers but is very easy to hit and has very bad saves. A wolf has pack tactics and smells and runs, but doesn't do much damage and doesn't have much hp. A bat has flight and blindsight and keen hearing, but can be dropped with one good hit.

**Comment by mlenser** [comment]
Moved Wild Shape to https://github.com/mlenser/kryx-rpg-issues/issues/285

**Comment by mlenser** [comment]
So I think Wild Shape is problematic due to https://github.com/mlenser/kryx-rpg-issues/issues/285, but lets take that discussion there.

Depending on the outcome of that we can then determine what to do here.

**Comment by marcosPlayX** [comment]
Thematically, these questions will also depend on what it means to change form. Are you becoming that thing literally, or just gaining its form and powers?

If you become that thing literally, then a bear doesn't cast spells, and a Fire Elemental only knows the spells it knows and casts it with Charisma (intrinsic power).

If the caster is just assuming that form, then maybe it should retain its own powers and cast with its standard spellcasting ability. A wizard that assumes the form of a Fire Elemental will be better or worse depending on how good or bad his own spellcasting ability is, since the power is not intrinsic to the Wizard, but simulated by their arcana. The more faithful a Cleirc, better is the storm elemental they can turn into.

**Comment by Marcloure** [comment]
Thematically, these questions will also depend on what it means to change form. Are you becoming that thing literally, or just gaining its form and powers?

If you become that thing literally, then a bear doesn't cast spells, and a Fire Elemental only knows the spells it knows and casts it with Charisma (intrinsic power).

If the caster is just assuming that form, then maybe it should retain its own powers and cast with its standard spellcasting ability. A wizard that assumes the form of a Fire Elemental will be better or worse depending on how good or bad his own spellcasting ability is, since the power is not intrinsic to the Wizard, but simulated by their arcana. The more faithful a Cleirc, better is the storm elemental they can turn into.

**Comment by itamarcu** [comment]
Personally I would prefer if Wild Shapes simply could not use spells that the spellcaster knows - or if they do, it should cost either extra resources or some high level feat. Maneuvers should potentially go the same way.

Not only is it much easier to balance each companion in a vacuum - I also just feel like Wild Shape is meant to be a transformation from your main form to a different form, and not a simple buff. Transformations need to have actual drawbacks - you gain the powers of your new form but you lose the powers of your old form. Compare it to Wind Walk, Gaseous Form, Polymorph - they have serious drawbacks that prevent the character from using their normal abilities while in the special form.

**Comment by mlenser** [comment]
> Are you becoming that thing literally, or just gaining its form and powers?

We discussed this last time we made a change to Wild Shape and the answer is becoming the thing. Otherwise Bestial Transformation exists.

But there are some limitations on mana/stamina and now HP/HD for thematic and balance reasons.

________________

> Transformations need to have actual drawbacks - you gain the powers of your new form but you lose the powers of your old form. Compare it to Wind Walk, Gaseous Form, Polymorph - they have serious drawbacks that prevent the character from using their normal abilities while in the special form.

I think Wild Shape and other transformations are different than these spells. Wild Shape should have some drawbacks, but you are still you (by retaining your Int/Wis/Cha). In RAW you can still maintain concentration on a spell and still retain all your features (including spells). You just can't use spells while transformed until 18th level in RAW.


______________

### Spellcasting

I had a thought earlier today: just make spellcasting in the new form a feat. It's a bit of a tax, but flying in the air and casting spells is very powerful so the tax is needed.

It should use the caster's normal spellcasting, not the monsters.

**Comment by itamarcu** [comment]
I agree, a feat should make this work well enough! I think the feat should be 8th or 12th level (when the druid might be able to obtain "normal" wings), but I don't know enough to say for sure.

How about maneuvers and concoctions? Would there be similar limits on them?

I think concoctions should definitely be limited (as they allow ranged attacks), and maneuvers... I don't know. Full-caster druids can't get maneuvers by leveling up (IIRC) so I wouldn't want the wild shapes to be balanced around the player being expected to add extra maneuver options and dice to the wild shape.

**Comment by mlenser** [comment]
I think the feat should be 4th level, which is the same as flying is restricted by RAW. Naturalists can then use it for the shapes that they can get at 2nd level and other classes can use it at 5th level when they can get the spells. This aligns with the other Fly options which is around 5th level, but has the cost of a feat so I believe the level is appropriate.

Concoction creation requires manual manipulation with hands so I'd say that's off the table. A bear could maybe grab a potion and drink it - I'll leave that up to the GM.

Maneuvers should be fully allowed.

**Comment by itamarcu** [comment]
Other flight options require concentration, though (unless you count Webshooter/vines) - this Wild Shape doesn't, which is an important benefit.

**Comment by mlenser** [comment]
The feat doesn't give flight. That's part of the base kit.

There is a connection there, but I don't think this should cost more than a feat. We already have other cases of feats improving a specific spell. The comparison on #285 is pretty balanced with the cost of a feat, imo. Sure, Wild Shape has some benefits, but realistically for the spell case case we're looking at:

### Wild Shape (Blood Hawk)
- Fly 60 feet for 2 hours
- +12 temporary hit points for 2 hours
- AC either the same or reduced by 1 or 2 if you have high AC (it's 14)
- Advantage on Perception that relies on sight
- (COST) 1 feat

### Levitate (Fly)
- Fly 60 feet for 10 minutes

### False Life
- ~22 temporary hit points for 2 hours

**Comment by itamarcu** [comment]
This seems alright to me. I'm just slightly concerned about the fact that this allows someone to simultaneously fly and cast offensive concentration spells (e.g. Animate Flames or Wall of Force), giving  them the ability to contribute to the fight with powerful spells while keeping themselves out of harm's way.

Without this, the only other way to do it (without another spellcaster) is to take feats that grant permanent flight, which appear at tier 2 or 3. Hence, I suggested making it higher level. I might be worried about it too much though (and the fact that dealing just a bit of damage to you can make you drop from the air is still a big weakness).

**Comment by mlenser** [comment]
- **Themes**
  - Spells that let you change form like Wild Shape, Tree Shape, Fire Form, etc have had their wording rewritten to clarify and simply. Additionally, spellcasting is not allowed while transformed. [#277](https://github.com/mlenser/kryx-rpg-issues/issues/277)
  - **Beast**
    - Beast Spells added as a 6th level feat to cast spells while transformed via Wild Shape [#277](https://github.com/mlenser/kryx-rpg-issues/issues/277)
  - **Plants**
    - Tree Spells added as a 6th level feat to cast spells while transformed via Tree Shape [#277](https://github.com/mlenser/kryx-rpg-issues/issues/277)

**Comment by Paulorpribeiro** [comment]
You are missing a talent or something to cast in elemental form I think. Also, I believe the elemental forms should be able to cast their theme spells (the spell in their companion stat block) by default. A lot of the power of the fire elemental form is being able to cast spells and apply burn for example

**Comment by Marcloure** [comment]
They already deal as much or more damage than other forms. Fire elemental can pass through creatures, ignite them on hit, deals the same 2d8+4 damage at 3 mana, so on. I don't think they need spells on top of it, otherwise elementals are much, much better.

**Comment by Paulorpribeiro** [comment]
Ah, ok, I didnt realize they ignite on hit now. 
Either way, there is a feat missing to allow caster to use spells in elemental form. They can't use the beast or the plant feat.

**Comment by mlenser** [comment]
I'm probably going to remove the elemental forms. That's why I didn't add feats yet.
